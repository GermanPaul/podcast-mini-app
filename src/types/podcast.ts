export interface TopPodcasts {
  feed: Feed
}

export interface Feed {
  author: Author
  entry: Entry[]
  updated: Updated
  rights: FeedRights
  title: FeedTitle
  icon: Icon
  link: FeedLink[]
  id: FeedId
}

export interface Author {
  name: Name
  uri: Uri
}

export interface Name {
  label: string
}

export interface Uri {
  label: string
}

export interface Entry {
  'im:name': ImName
  'im:image': ImImage[]
  summary: Summary
  'im:price': ImPrice
  'im:contentType': ImContentType
  rights: Rights
  title: Title
  link: Link
  id: Id
  'im:artist': ImArtist
  category: Category
  'im:releaseDate': ImReleaseDate
}

export interface ImName {
  label: string
}

export interface ImImage {
  label: string
  attributes: ImImageAttributes
}

export interface ImImageAttributes {
  height: string
}

export interface Summary {
  label: string
}

export interface ImPrice {
  label: string
  attributes: PriceAttributes
}

export interface PriceAttributes {
  amount: string
  currency: string
}

export interface ImContentType {
  attributes: ContentTypeAttributes
}

export interface ContentTypeAttributes {
  term: string
  label: string
}

export interface Rights {
  label: string
}

export interface Title {
  label: string
}

export interface Link {
  attributes: LinkAttributes
}

export interface LinkAttributes {
  rel: string
  type: string
  href: string
}

export interface Id {
  label: string
  attributes: IdAttributes
}

export interface IdAttributes {
  'im:id': string
}

export interface ImArtist {
  label: string
  attributes: ArtistAttributes
}

export interface ArtistAttributes {
  href: string
}

export interface Category {
  attributes: CategoryAttributes
}

export interface CategoryAttributes {
  'im:id': string
  term: string
  scheme: string
  label: string
}

export interface ImReleaseDate {
  label: string
  attributes: ReleaseDateAttributes
}

export interface ReleaseDateAttributes {
  label: string
}

export interface Updated {
  label: string
}

export interface FeedRights {
  label: string
}

export interface FeedTitle {
  label: string
}

export interface Icon {
  label: string
}

export interface FeedLink {
  attributes: Attributes9
}

export interface Attributes9 {
  rel: string
  type?: string
  href: string
}

export interface FeedId {
  label: string
}
