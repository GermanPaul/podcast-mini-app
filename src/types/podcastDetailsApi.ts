export interface PodcastDetails {
  title: string
  author: string
  description: string
  image: string
  items: PodcastDetailsItem[]
}

export interface PodcastDetailsItem {
  id: string
  title: string
  date: string
  duration: string
  description: string
  url: string
}

export interface RawPodcastDetails {
  title: string;
  description: string;
  image?: { url: string };
  'itunes:author': string;
  'itunes:image': { href: string }
  item: RawPodcastDetailsItem[]
}

export interface RawPodcastDetailsItem {
  title: string;
  pubDate: string;
  'itunes:duration'?: string;
  description: string;
  enclosure: { url: string }
  guid: { '#text': string } | string
}

export interface PodcastDetailsApiResponse {
  resultCount: number
  results: Result[]
}

export interface Result {
  wrapperType: string
  kind: string
  artistId: number
  collectionId: number
  trackId: number
  artistName: string
  collectionName: string
  trackName: string
  collectionCensoredName: string
  trackCensoredName: string
  artistViewUrl: string
  collectionViewUrl: string
  feedUrl: string
  trackViewUrl: string
  artworkUrl30: string
  artworkUrl60: string
  artworkUrl100: string
  collectionPrice: number
  trackPrice: number
  collectionHdPrice: number
  releaseDate: string
  collectionExplicitness: string
  trackExplicitness: string
  trackCount: number
  trackTimeMillis: number
  country: string
  currency: string
  primaryGenreName: string
  contentAdvisoryRating: string
  artworkUrl600: string
  genreIds: string[]
  genres: string[]
}
