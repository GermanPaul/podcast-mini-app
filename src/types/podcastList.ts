export interface FetchPodcastFeedInput {
  limit: number
  genre: number
}

export interface PodcastFeed {
  id: string
  name: string
  author: string
  image: string
}
