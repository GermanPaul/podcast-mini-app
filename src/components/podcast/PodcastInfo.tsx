import React from 'react'
import { PodcastDetails } from '@/types/podcastDetailsApi'
import { useI18n } from '@/hooks/useI18n'

interface PodcastInfoProps {
  podcast: PodcastDetails
}

const PodcastInfo = ({ podcast }: PodcastInfoProps) => {
  const { t } = useI18n()

  return <div className='p-4 shadow-md'>
    <img className='mx-auto mb-4 rounded-md w-72 h-72' src={podcast.image} alt={podcast.title} loading='lazy' />
    <div className='my-3 mx-4'>
      <div className='font-bold'>{podcast.title}</div>
      <div className='italic'>{t('PODCAST_BY')} {podcast.author}</div>
    </div>
    <div className='my-3 mx-4'>
      <div className='font-bold'>{t('PODCAST_DESCRIPTION')}</div>
      <div dangerouslySetInnerHTML={{__html: podcast.description}}></div>
    </div>
  </div>
}

export default PodcastInfo
