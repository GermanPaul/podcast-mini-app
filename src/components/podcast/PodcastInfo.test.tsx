import { render, screen } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import { PodcastDetails } from '@/types/podcastDetailsApi'

import PodcastInfo from './PodcastInfo'


const podcastDetails: PodcastDetails = {
  title: 'My first podcast',
  author: 'Rookie',
  description: 'Our first <b>podcast!</b>',
  image: 'https://image.provider/rookie/my-first-podcast.jpg',
  items: []
}

describe('<PodcastInfo />', () => {
  it('render the component', () => {
    const { baseElement } = render(<IntlProvider 
      locale='en' 
      messages={{ PODCAST_DESCRIPTION: 'Description', PODCAST_BY: 'By' }}
    >
      <PodcastInfo podcast={podcastDetails} />
    </IntlProvider>)

    expect(baseElement).toMatchSnapshot()
  })

  it('should display the podcast title', () => {
    render(<IntlProvider 
      locale='en' 
      messages={{ PODCAST_DESCRIPTION: 'Description', PODCAST_BY: 'By' }}
    >
      <PodcastInfo podcast={podcastDetails} />
    </IntlProvider>)

    expect(screen.getByTestId('podcast-title')).toHaveTextContent('My first podcast')
  })

  it('should display the podcast author', () => {
    render(<IntlProvider 
      locale='en' 
      messages={{ PODCAST_DESCRIPTION: 'Description', PODCAST_BY: 'By' }}
    >
      <PodcastInfo podcast={podcastDetails} />
    </IntlProvider>)

    expect(screen.getByTestId('podcast-author')).toHaveTextContent('Rookie')
  })

  it('should display the podcast description as HTML', () => {
    render(<IntlProvider 
      locale='en' 
      messages={{ PODCAST_DESCRIPTION: 'Description', PODCAST_BY: 'By' }}
    >
      <PodcastInfo podcast={podcastDetails} />
    </IntlProvider>)

    expect(screen.getByTestId('podcast-description')).toContainHTML('Our first <b>podcast!</b>')
  })

  it('should load the podcast image', () => {
    const { container } = render(<IntlProvider 
      locale='en' 
      messages={{ PODCAST_DESCRIPTION: 'Description', PODCAST_BY: 'By' }}
    >
      <PodcastInfo podcast={podcastDetails} />
    </IntlProvider>)

    expect(container.querySelector('img')).toHaveProperty('src', 'https://image.provider/rookie/my-first-podcast.jpg')
  })
})
