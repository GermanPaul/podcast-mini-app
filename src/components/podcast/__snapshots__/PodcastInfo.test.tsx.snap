// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PodcastInfo /> > render the component 1`] = `
<body>
  <div>
    <div
      class="p-4 shadow-md"
    >
      <img
        alt="My first podcast"
        class="mx-auto mb-4 rounded-md w-72 h-72"
        loading="lazy"
        src="https://image.provider/rookie/my-first-podcast.jpg"
      />
      <div
        class="my-3 mx-4"
      >
        <div
          class="font-bold"
          data-testid="podcast-title"
        >
          My first podcast
        </div>
        <div
          class="italic"
          data-testid="podcast-author"
        >
          By
           
          Rookie
        </div>
      </div>
      <div
        class="my-3 mx-4"
      >
        <div
          class="font-bold"
        >
          Description
        </div>
        <div
          data-testid="podcast-description"
        >
          Our first 
          <b>
            podcast!
          </b>
        </div>
      </div>
    </div>
  </div>
</body>
`;
